app/models/manageiq/providers/lenovo/physical_infra_manager/parsers/components/firmware_parser.rb

```diff
@@ -0,0 +1,25 @@
+require_relative 'component_parser'
```

Comment:
Don't use require or require_relative as we should allow Rails to use its built-in autoload functionality.  The main reason is that in dev mode, Rails will reload classes/constants from disk on source change, and using require/require_relative prevents that, which can actually break things.   Additionally, the class in this file does not match the file path, which breaks Rails autoload.

Reply:
Done by rebasing with @agrare 's PR.